### PR TITLE
Add initial tests for utility and configure CI with pytest

### PR DESCRIPTION
**Pull request by Borda**
This pull request introduces automated testing infrastructure and adds comprehensive unit tests for the `SmoothedValue` utility. It also updates the project dependencies to include required testing packages.

**CI/CD and Testing Infrastructure:**

* Adds a new GitHub Actions workflow (`.github/workflows/ci-tests.yml`) to run pytest-based tests across multiple operating systems (Ubuntu, Windows, macOS) and Python versions (3.9–3.13), including coverage reporting and integration with Codecov. Also introduces a "testing guardian" job to ensure test failures or cancellations are surfaced clearly.

**Dependency Management:**

* Updates `pyproject.toml` to add a `[project.optional-dependencies] tests` group, specifying `pytest` and `pytest-cov` as testing dependencies.

**Unit Tests:**

* Adds a new test module `tests/util/test_misc.py` with comprehensive parametric and edge-case tests for the `SmoothedValue` class, covering initialization, update logic, property calculations, string formatting, exception handling, and distributed synchronization behavior.

### CODE REVIEW

File: .github/workflows/ci-tests.yml [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+  contents: read
+
+concurrency:
+  group: pytest-test-${{ github.ref }}
+  cancel-in-progress: ${{ github.event_name == 'pull_request' }}
+
+jobs:
+  build-pkg:
+    uses: ./.github/workflows/build-package.yml
+
```
[the remark this answers is not in the document]
Comment (Borda) [reply .github/workflows/ci-tests.yml]: we want to test also build a package, but maybe we can do it in the build workflow...

File: tests/util/test_misc.py [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+        (20, None),
+        (5, "{median:.1f} ({global_avg:.1f})"),
+    ])
+    def test_init(self, window_size, fmt):
+        if fmt is None:
+            sv = SmoothedValue(window_size=window_size)
+            assert sv.fmt == "{median:.4f} ({global_avg:.4f})"
+        else:
+            sv = SmoothedValue(window_size=window_size, fmt=fmt)
+            assert sv.fmt == fmt
```
Comment (SkalskiP) [inline tests/util/test_misc.py]: do we need to make those two separate test cases? If `fmt` is `None` we can still pass it as `SmoothedValue(window_size=window_size, fmt=fmt)`.

File: tests/util/test_misc.py [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+    ])
+    def test_empty_exceptions(self, property_name, expected_exception):
+        sv = SmoothedValue()
+        with pytest.raises(expected_exception):
+            getattr(sv, property_name)
+
+    @pytest.mark.parametrize("property_name", ["median", "avg"])
+    def test_empty_nan(self, property_name):
+        sv = SmoothedValue()
+        assert math.isnan(getattr(sv, property_name))
```
Comment (SkalskiP) [inline tests/util/test_misc.py]: We could probably doin raising and non raising cases. We ofted do that in supervision with `from contextlib import ExitStack as DoesNotRaise`. Take a look [here](https://github.com/roboflow/supervision/blob/82a0aa2463fa0e0d09181194f5b84324285fe819/test/annotators/test_utils.py).

File: .github/workflows/ci-tests.yml [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+    strategy:
+      fail-fast: false
+      matrix:
+        os: ["ubuntu-latest"]
+        python-version: ["3.9", "3.10", "3.11", "3.12", "3.13"]
+        include:
+          - { os: "windows-latest", python-version: "3.9" }
+          - { os: "macos-latest", python-version: "3.9" }
+          - { os: "windows-latest", python-version: "3.13" }
+          - { os: "macos-latest", python-version: "3.13" }
```
[the remark this answers is not in the document]
Comment (Borda) [reply .github/workflows/ci-tests.yml]: So when you merge it, I will update it here

File: tests/util/test_misc.py (hunk 2 above)
Comment (Borda) [reply tests/util/test_misc.py]: simplified it
